Nef_3/include/CGAL/Nef_3/SNC_k3_tree_traits.h

```diff
@@ -417,7 +417,6 @@ Side_of_plane<SNC_decorator>::operator()
 #ifdef CGAL_NEF_EXPLOIT_REFERENCE_COUNTING
   }
 #endif
-  CGAL_error_msg( "should not be reached");
```

Comment:
I do not like that change. The `CGAL_error_msg`.

The proper patch would be to qualify `CGAL::assertion_fail` as [`__declspec(noreturn)`][doc]. All versions of MSVC support that attribute. The macro `CGAL_NORETURN` defined here:
https://github.com/CGAL/cgal/blob/a7a4966de82c48df461ed556e27582f527d7b7e2/Installation/include/CGAL/config.h#L527-L532
should be updated, to that `__declspec(noreturn)` is used with MSVC.

[doc]: https://msdn.microsoft.com/en-us/library/k6ktzx3s(v=vs.110).aspx

The documentation is clear:
> This `__declspec` attribute tells the compiler that a function does not return. As a consequence, the compiler knows that the code following a call to a `__declspec(noreturn)` function is unreachable. 

If `CGAL::assertion_fail` is qualified with `noreturn`, then anything after `CGAL_error()` or`CGAL_error_msg("oups")` is unreachable.

Reply:
My remark about the patch is wrong. The `CGAL_error_msg` itself is unreachable.

I will open another issue to address my proposal modify `CGAL_NORETURN`.